src/frontend/src/handler/create_sink.rs

```diff
@@ -626,6 +627,10 @@ static CONNECTORS_COMPATIBLE_FORMATS: LazyLock<HashMap<String, HashMap<Format, V
                     Format::Plain => vec![Encode::Json,Encode::Template],
                     Format::Upsert => vec![Encode::Json,Encode::Template],
                 ),
+                "table" => hashmap!(
```

Comment:
FYI there is `SinkFormatDesc::from_legacy_type`. It would make sure that backend always [1] has access to `SinkFormatDesc` for **connectors opted-in** no matter frontend received old or new syntax from user. We may want to opt-in `connector = table` as well.

[1] New code paths in backend need to call `SinkFormatDesc::from_legacy_type` immediately in every new places a proto is decoded.

Reply:
forbid using the old syntax in the create sink into table in https://github.com/risingwavelabs/risingwave/pull/14134/commits/4f17ed315ead6c3c1f9411a3c42564fee5e135f1.